Large first step for sbgnviz viewer re-factoring, optimizing, simplifying - in the 'unstable' branch

IgorRodchenkov opened the pull request:
Once these changes get accepted and merged, I will go ahead to fix issue #165 and more.

IgorRodchenkov commented:
If you want to see the code, go to:

https://github.com/PathwayCommons/sbgnviz-js/tree/unstable 

and try here:

http://beta.pathwaycommons.org/sbgnviz/

('samples' directory is not in the 'unstable' branch, *.xml are actually ignored by git there; so I added samples manually for the app to work as normal, as that from 'master' branch)

IgorRodchenkov commented:
The main point of these changes was to properly use cytoscape.js (initialize cy, extensions, event handlers once; then reload network data only); [see this function](https://github.com/PathwayCommons/sbgnviz-js/blob/unstable/js/sbgnviz.js#L3).

Other changes and re-factoring were to optimise, somewhat simplify and cleanup the the viewer code, prepare for future making it npm module(s)...

metincansiper commented:
It seems that the reason for one of the comments above "('samples' directory is not in the 'unstable' branch, *.xml are actually ignored by git there; so I added samples manually for the app to work as normal, as that from 'master' branch)" is that "*.xml" is added to "gitignore". 

I think that this problem will disappear when we remove that line. Please see "https://github.com/iVis-at-Bilkent/chise.js/issues/85"

IgorRodchenkov commented:
This also fixes #172

metincansiper commented:
@IgorRodchenkov we decided that we need to have the same file structure in ChiSE and SBGNViz. Therefore I needed to seperate some files that you combined in this PR. I removed 'sbgnviz.js' file and distributed its content into 3 files 'general-sample-app-functions-and-variables.js', 'sample-app-menu-functions.js', 'sample-app-cytoscape-sbgn.js'. Also, I added 'add-remove-action-functions.js' back. I think that the number of files is just increased by 3.

Also I added the 'sample-app' folder back. However, I keep 'index.html' outside of it. In this way we can still run the application without having 'sample-app' in the path.

There were some bugs coming with this PR I fixed them  (Delete simple/smart was not working,
spinner was not working in operations other than performing layout, disconnected nodes was being
removed most probably this was not a bug but it is not something we expect in SBGN viewer).

Thanks for your contribution.

IgorRodchenkov commented:
Thanks for merging.
But.
'Delete' was not supposed to work in the VIEWER, I bet (hide and highlight is sufficient). I kinda dislike long unclear filenames... sadly.

Igor

> On Nov 1, 2016, at 7:08 AM, metincansiper <notifications@github.com> wrote:
> 
> @IgorRodchenkov we decided that we need to have the same file structure in ChiSE and SBGNViz. Therefore I needed to seperate some files that you combined in this PR. I removed 'sbgnviz.js' file and distributed its content into 3 files 'general-sample-app-functions-and-variables.js', 'sample-app-menu-functions.js', 'sample-app-cytoscape-sbgn.js'. Also, I added 'add-remove-action-functions.js' back. I think that the number of files is just increased by 3.
> 
> Also I added the 'sample-app' folder back. However, I keep 'index.html' outside of it. In this way we can still run the application without having 'sample-app' in the path.
> 
> There were some bugs coming with this PR I fixed them (Delete simple/smart was not working,
> spinner was not working in operations other than performing layout, disconnected nodes was being
> removed most probably this was not a bug but it is not something we expect in SBGN viewer).
> 
> Thanks for your contribution.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

IgorRodchenkov commented:
BTW, disconnected nodes were being removed before my PR; in master branch too (see SBGNContainer.render there)

Igor

> On Nov 1, 2016, at 7:08 AM, metincansiper <notifications@github.com> wrote:
> 
> @IgorRodchenkov we decided that we need to have the same file structure in ChiSE and SBGNViz. Therefore I needed to seperate some files that you combined in this PR. I removed 'sbgnviz.js' file and distributed its content into 3 files 'general-sample-app-functions-and-variables.js', 'sample-app-menu-functions.js', 'sample-app-cytoscape-sbgn.js'. Also, I added 'add-remove-action-functions.js' back. I think that the number of files is just increased by 3.
> 
> Also I added the 'sample-app' folder back. However, I keep 'index.html' outside of it. In this way we can still run the application without having 'sample-app' in the path.
> 
> There were some bugs coming with this PR I fixed them (Delete simple/smart was not working,
> spinner was not working in operations other than performing layout, disconnected nodes was being
> removed most probably this was not a bug but it is not something we expect in SBGN viewer).
> 
> Thanks for your contribution.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

ugurdogrusoz commented:
@IgorRodchenkov 

Delete is different than Hide since it permanently removes nodes/edges, whereas Show All will bring back all hidden nodes/edges. Users might load and prefer to do analysis after getting rid of unneeded parts of a pathway with Delete.

Disconnected nodes should be removed by the user (in your case your viewer); we cannot assume all users of SBGNViz will prefer to remove them on load.

@metincansiper will take a look at file names to see if there can be improvements with the names but I think long file names are OK :)

Thanks a lot again for your contribution, highly appreciated.

metincansiper commented:
I revised some file names and replied the related comment (https://github.com/iVis-at-Bilkent/sbgnviz-js/commit/b8815935986cb5742bb922f8909b93e24cd35240#commitcomment-19658176)
